app/models/repo.js

```diff
@@ -68,7 +69,23 @@ const Repo = VcsEntity.extend({
     return hasPermissions && (!isMigrated || isFailed);
   }),
 
-  allowance: reads('owner.allowance'),
+  allowance: computed('owner.allowance', 'repoOwnerAllowance', function () {
+    if (this.owner.allowance)
+      return this.owner.allowance;
+    if (this.repoOwnerAllowance)
+      return this.repoOwnerAllowance;
+    this.fetchRepoOwnerAllowance.perform();
+  }),
+
+  repoOwnerAllowance: reads('fetchRepoOwnerAllowance.lastSuccessful.value'),
+
+  fetchRepoOwnerAllowance: task(function* () {
+    const allowance = this.store.peekRecord('allowance', this.owner.id);
+    if (allowance)
+      return allowance;
+    yield this.store.queryRecord('allowance', { login: this.owner.login, provider: this.provider });
```

Comment:
The `yield`ed value needs to be returned in order to get set as task's value:
```suggestion
    return yield this.store.queryRecord('allowance', { login: this.owner.login, provider: this.provider });
```
Currently, the `fetchRepoOwnerAllowance.lastSuccessful.value` will be undefined after fetch.